feat(data): Creating Unclassified and Unclassified Minor Categories

Zitchas opened the pull request:
**Content**

When doing analysis of our overall ship set, there's a whole variety of ships that are basically just dumped into categories because they need a category; or otherwise don't really fit within the player's paradigm. As in, these are ships that are never acquirable in any vanilla in-game way (~~the Shooting Star being the exception~~), aren't manufactured or produced conventionally speaking, and generally aren't representative of the categories they've been placed in. 

This creates two categories: `"Unclassified"` and `"Unclassified Minor"`

The former is the primary category that virtually everything was placed into; while the latter is for any of these unclassified things that were previously designated fighters, drones, infants, or otherwise something that suggests they might be carrieable.

Basically, this affects the Void Sprites and all the rest of the space creatures and a few of the person ships that aren't identifiable manufactured ships. 

Yes, I have included the Archon in this list, because a) it's a placeholder, b) we don't actually know what it is, c) it's not obvious to the player that it is even manufactured, or d) it's true purpose. And at the end of the day, it just messes up the tables to have this basically invincible ship tacked on to the list. 

## Gameplay Impact:

~~With the exception of the Shooting Star,~~ This is generally not a directly in-game visible change.

It is, however, highly visible when using an all-content plug-in, the Endless Sky Encyclopedia, spreadsheets, or wikis; which all generally use these categories. As such, for those that discover this piece of information about these ships, I feel that listing them as "Unclassified" makes them more mysterious and less misleading; and thus helps to reinforce their nature.

## Testing Done
Played the game. Tested that the Ayym can still pick carry, launch, and recover Jje.

## Save File
Any pilot, anytime, anywhere; but effects will generally only visible when using an all-content plugin

## Performance Impact
none

Edti: By BeccaBunny's request, the Shooting Star category has been reverted to "Heavy Warship." as being deliberate for entertainment value.

beccabunny commented:
I disagree with changing the shooting star's category, it was put like that intentionally simply because I found the concept of a space rock classified as a heavy warship amusing

The other edits make sense to me

Zitchas commented:
Sure, I can change that.

warp-core commented:
I don't think the given reasoning really makes sense for recategorising the Ursa Polaris.
Sure, it might look somewhat unique for a space ship, but that's not really how categories work.
Its stats aren't really any more different to any normal ship than any other person ship.

There are also several "ship"s in the game that don't have any category. If we're adding an "N?A" type category for stuff that seems a bit odd in the normal categories, it seems reasonable to also put those category-less ships into it as well.

Saugia commented:
I would add the vyu-Ir to this list as it's intended to be an unknown Ruleian ship - could have been a HW, could have been a LW, could have been some Utility ship, etc..

Zitchas commented:
OK, I've added the Hallucination, Asteroid, and vyu-Ir. I wasn't sure about the lore behind that one, thanks for clarifying.

This leaves the two timer ships, the nanobots, and the rescue dummy. I'm reluctant to include these three because they're not ships at all, they're not even pretend things that are kind of similar to ships. They're just flat out mechanics. 

Honestly, if these were to be given a category, I'd want to label them "Game Mechanics." If people are interested in having that, I'm happy to oblige here.

Regarding Ursa Polaris, I was actually more inclined to go through the Persons list and label anything that wasn't an actual (no attributes modified) variant of a real ship as unclassified, thus limiting the actual categories to things that actual fall within the game's normal framework. If it's sufficiently disruptive that people want to remove it from the game, it probably isn't representative of its category anymore.

As to Ursa in particular, while it does not have egregious stats, it's not purchasable, it's not acquirable, it's not part of any in-game lore anywhere, it doesn't even match any faction aesthetics. The fact that it's a "heavy warship" that has a cargo capacity matched only by a handful of Superheavy's just doesn't really jive. In any case, the point isn't to argue the Ursa Polaris's stats, the point is to set it outside the mainstream categories so it is clear it *is* outside them, and not just an oddball outlier in the Heavy Warship category that might be subject to correction or rebalancing in the future. 


Anyway, if people would like the game mechanics ships in a "Game Mechanics" category, just say so, and I'll do it.

bene-dictator commented:
> Honestly, if these were to be given a category, I'd want to label them "Game Mechanics." If people are interested in having that, I'm happy to oblige here.

I'd hesitate to do that since I think we should want to have as few "ships" doing these roles as possible. I'd keep the nanobots and the Remnant rescue dummy as unclassified.

Zitchas commented:
Oh, definitely. If we had the mechanics to fully replicate what those timers ships do, I'd replace them in an instant. Categorizing them as "Game Mechanics" could be just a means of collecting them as a group of ships that exist exclusively for the purpose of game mechanics, nothing more.

warp-core commented:
> As to Ursa in particular, while it does not have egregious stats, it's not purchasable, it's not acquirable, it's not part of any in-game lore anywhere, it doesn't even match any faction aesthetics. The fact that it's a "heavy warship" that has a cargo capacity matched only by a handful of Superheavy's just doesn't really jive. (the highest cargo capacity of any Heavy Warship the player might conceivably get their hands on is the Ra'at Ik 621 with 288) In any case, the point isn't to argue the Ursa Polaris's stats, the point is to set it outside the mainstream categories so it is clear it is outside them, and not just an oddball outlier in the Heavy Warship category that might be subject to correction or rebalancing in the future.

Many of these characteristics are also true for other person ships. Usually, it's the outfit and weapon space that is abnormally high. This one happens to have a high cargo space value as well. There are plenty of other categories that this could go in if that's the problem. I don't think this is otherwise sufficiently different from other person ships to be categorised so differently from them.

MasterOfGrey commented:
Worth noting that that if the Ursa Polaris is going to remain in the regular ship categories, it should be in the Utility class.

mOctave reviewed (approved):
Looks good! My only concern is that Unclassified Minor is a bit of a confusing name. Maybe "Unclassified Carried"?

bene-dictator commented:
> Categorizing them as "Game Mechanics" could be just a means of collecting them as a group of ships that exist exclusively for the purpose of game mechanics, nothing more.

It seems to me that this class would just be temporary if we plan to replace them; I don't see the need for another classification in that case, if it's for all of four ships. And if #9089 is merged (provided a dev can get their priorities in order and finish off the multitude of PRs that have been laying around), hopefully two.

Zitchas commented:
> It seems to me that this class would just be temporary if we plan to replace them;

While I have said I would love to replace them, there is currently no indication anywhere that anyone considers this a priority, or even on the "to-do list" at all. I know it's currently beyond the scope of my ability to do easily, and it isn't even really on my priority list at all, either. It's something I'd like to see happen, but not enough that I'm going to put effort into it.

Case in point, the linked PR hasn't exactly been pushed along with energy. In any case, changing a ship category when the ship is deleted in favor of an actual game mechanic is not exactly a problem. It's not adding work or anything.

Quantumshark commented on data/drak/indigenous.txt:
```diff
@@ -125,7 +125,7 @@ ship "Void Sprite" "Void Sprite (Infant)"
 	"never disabled"
 	attributes
-		category "Light Warship"
+		category "Unclassified Minor"
```
This isn't a carried unit, so should probably be `unclassified` rather than `unclassified minor`.

Zitchas replied:
It's an infant, which is to say minor, void sprite. Even if there's currently no void sprites that carry it, it still felt appropriate in terms of lore to name it as such.

Saugia replied:
Categories are meant to be mechanical terms of a ship's purpose - i.e., utility, heavy warship, etc., - they aren't necessarily a reference to a ship's lore, but to its functionality. So in this case, for mechanical consistency, if we are to have a `minor` tag for `unclassified`, it makes sense that it should be tied to a difference in function, not lore, which would be the ability to be carried in this case.

Saugia replied:
If `unclassified minor` is meant to be specifically relative to a ships' (or species') size or growth compared to others, it would make more sense that there is an overall notable amount of lifeforms like this, though I don't think there are enough to warrant that distinction, and ultimately, the `minor` is necessary because the carriable Jje need to be a separate category anyways.

Otherwise, it makes sense that any non "-ship" would just be `Unclassified`. It seems odd to have multiple types of "Unclassification" for something that is a result of ambiguity - if we saw to different aliens in real life, say, one large. "old" and large in numbers, and one small and "young", I'd imagine we just call them "aliens", not "alien minor" and "alien"; the `minor` designation is more a result of a necessary functionality.

Quantumshark commented on data/korath/nanobots.txt:
```diff
@@ -13,6 +13,7 @@
 
 ship "Nanobot"
 	attributes
+		category "Game Mechanics"
```
```suggestion
		category "Unclassified"
```
The nanobot is, in-universe, a thing (albeit evidently representing a swarm of nanobots rather than a single one) rather than an out-of-universe game mechanic, so unclassified is probably more appropriate.

ziproot commented on data/categories.txt:
```diff
@@ -25,12 +25,16 @@ category "ship"
 	"Fighter"
 	"Drone"
+	"Unclassified"
+	"Unclassified Minor"
+	"Game Mechanics"
```
```suggestion
```

Saugia replied:
I still think there's reason to separate any ships specifically made for a game mechanic to operate, as opposed to ships that exist within universe and are able to be classified or not. These are two very different things - one is code/mechanic based, one is purpose-based within the game.

ziproot replied:
Updated

ziproot commented on data/drak/indigenous.txt:
```diff
@@ -125,7 +125,7 @@ ship "Void Sprite" "Void Sprite (Infant)"
 	"never disabled"
 	attributes
-		category "Light Warship"
+		category "Unclassified Minor"
```
```suggestion
		category "Unclassified"
```

ziproot commented on data/categories.txt:
```diff
@@ -25,12 +25,16 @@ category "ship"
 category "bay type"
 	"Drone"
 	"Fighter"
+	"Unclassified Minor"
```
```suggestion
	"Unclassified (Carried)"
```

ziproot commented on data/iije/iije.txt:
```diff
@@ -21,7 +21,7 @@ ship "Jje"
 	"never disabled"
 	attributes
-		category "Fighter"
+		category "Unclassified Minor"
```
```suggestion
		category "Unclassified (Carried)"
```

ziproot commented on data/iije/iije.txt:
```diff
@@ -123,31 +123,31 @@ ship "Ayym"
 	engine 0 0
 	"steering engine" 0 0
-	bay "Fighter" -36 40
+	bay "Unclassified Minor" -36 40
```
```suggestion
	bay "Unclassified (Carried)" -36 40
```

ziproot commented on data/iije/iije.txt:
```diff
@@ -123,31 +123,31 @@ ship "Ayym"
 		over
 		angle 60
-	bay "Fighter" 36 20
+	bay "Unclassified Minor" 36 20
```
```suggestion
	bay "Unclassified (Carried)" 36 20
```

ziproot commented on data/iije/iije.txt:
```diff
@@ -123,31 +123,31 @@ ship "Ayym"
 		over
 		angle -70
-	bay "Fighter" -16 104
+	bay "Unclassified Minor" -16 104
```
```suggestion
	bay "Unclassified (Carried)" -16 104
```

ziproot commented on data/iije/iije.txt:
```diff
@@ -123,31 +123,31 @@ ship "Ayym"
 		over
 		angle 162
-	bay "Fighter" -28 66
+	bay "Unclassified Minor" -28 66
```
```suggestion
	bay "Unclassified (Carried)" -28 66
```

ziproot commented on data/iije/iije.txt:
```diff
@@ -123,31 +123,31 @@ ship "Ayym"
 		over
 		angle 68
-	bay "Fighter" -18 14
+	bay "Unclassified Minor" -18 14
```
```suggestion
	bay "Unclassified (Carried)" -18 14
```

ziproot commented on data/iije/iije.txt:
```diff
@@ -123,31 +123,31 @@ ship "Ayym"
 		over
 		angle 127
-	bay "Fighter" 35 25
+	bay "Unclassified Minor" 35 25
```
```suggestion
	bay "Unclassified (Carried)" 35 25
```

ziproot commented on data/iije/iije.txt:
```diff
@@ -123,31 +123,31 @@ ship "Ayym"
 		over
 		angle 25
-	bay "Fighter" 25 70
+	bay "Unclassified Minor" 25 70
```
```suggestion
	bay "Unclassified (Carried)" 25 70
```

ziproot commented on data/iije/iije.txt:
```diff
@@ -123,31 +123,31 @@ ship "Ayym"
 		over
 		angle -120
-	bay "Fighter" 11 110
+	bay "Unclassified Minor" 11 110
```
```suggestion
	bay "Unclassified (Carried)" 11 110
```

ziproot commented on data/iije/iije.txt:
```diff
@@ -123,31 +123,31 @@ ship "Ayym"
 		over
 		angle 38
-	bay "Fighter" -36 40
+	bay "Unclassified Minor" -36 40
```
```suggestion
	bay "Unclassified (Carried)" -36 40
```

ziproot commented on data/remnant/remnant 1 introduction.txt:
```diff
@@ -3655,6 +3655,7 @@ ship "_Ion Timer Ship"
 	# The _Ion Storm Timer: Generator outfit counteracts this until there is an ion storm
 	# sprite "ship/shuttle" # uncomment this for debugging
 	attributes
+		category "Game Mechanics"
```
```suggestion
		category "Unclassified"
```

Saugia replied:
This is explicitly used for the mechanic of the ion storm to operate. I wouldn't say this is "unclassified", it is quite literally used for the function. If we are using something that is to operate a mechanic, it is a game mechanic in classification, despite it being an "unclassified" natural body.

ziproot commented on data/persons.txt:
```diff
@@ -788,7 +788,7 @@ ship "Ursa Polaris"
 	sprite "ship/localworldship"
 	attributes
-		category "Heavy Warship"
+		category "Unclassified"
```
```suggestion
		category "Utility"
```

Saugia replied:
I do agree with this though - the Ursa Polaris isn't the same thing as Becca's asteroid - it is still a ship with a constructed body, even if that's partly with an asteroid. It makes sense it either keeps "Heavy Warship, as it is a ship that battles, or "Utility", if we are considering an Asteroid being a base. It's still a "-ship" either way", so I wouldn't say it's "unclassified".

ziproot commented on data/remnant/remnant jobs.txt:
```diff
@@ -383,6 +383,7 @@ mission "Remnant: Rescue"
 ship "Rescue Dummy"
 	# sprite "ship/behemoth"
 	attributes
+		category "Game Mechanics"
```
```suggestion
		category "Unclassified"
```

ziproot commented on data/sheragi/archaeology missions.txt:
```diff
@@ -300,6 +300,7 @@ mission "Deep Archaeology 5"
 
 ship "Timer Ship"
 	attributes
+		category "Game Mechanics"
```
```suggestion
		category "Unclassified"
```

Saugia replied:
Same as with the Ion Storm, this is explicitly used for a mechanic, not a ship classification.

ziproot commented on data/vyrmeid/vyrmeid.txt:
```diff
@@ -20,7 +20,7 @@ ship "Vyrmeid"
 	"never disabled"
 	attributes
-		"category" "Fighter"
+		"category" "Unclassified Minor"
```
```suggestion
		"category" "Unclassified (Carried)"
```

ziproot commented on data/vyrmeid/vyrmeid.txt:
```diff
@@ -77,7 +77,7 @@ ship "Vyrmeid" "Vyrmeid (B)"
 	"never disabled"
 	attributes
-		"category" "Fighter"
+		"category" "Unclassified Minor"
```
```suggestion
		"category" "Unclassified (Carried)"
```

ziproot commented on data/vyrmeid/vyrmeid.txt:
```diff
@@ -128,7 +128,7 @@ ship "Vyrmeid" "Vyrmeid (C)"
 	"never disabled"
 	attributes
-		category "Fighter"
+		category "Unclassified Minor"
```
```suggestion
		category "Unclassified (Carried)"
```

ziproot replied:
```suggestion
		category "Unclassified (Carried)"
```

ziproot reviewed (commented):
I prefer unclassified (carried) as it is easier to understand the purpose behind it and it doesn't lead to confusion with ships such as the infant void sprite, which cannot be carried. I also don't think as many categories are necessary. All "unclassified" ships are ships that you don't actually use in game. I would be fine with this PR being merged as-is, however, I would prefer if these changes were made.

MasterOfGrey commented:
I'll second the swap to `"Unclassified (Carried)"`

Zitchas commented:
To be honest, the point of "Unclassified" is to be *less* specific, not more so. Just calling it "Carried" sounds so... normal? plain? generic?

minor doesn't specify anything clearly, although it allows us all to assume various things. 

If it was changed to something else, I'd rather "Alternate Unclassified", "Different Unclassified", or something.

Actually, perhaps the best option might be "Unclassified " (as in only different from the other name by the addition of a space; so it won't appear to actually have a different name)

ziproot commented:
Yeah, I was considering just having both options be “unclassified” but I wasn’t sure if that would be too confusing. If you think that is a better option, you can go for it. I don’t think the extra space is necessary, as the current implementation uses “unclassified minor” as both a carriable and a parent category.

Saugia commented:
> Actually, perhaps the best option might be "Unclassified " (as in only different from the other name by the addition of a space; so it won't appear to actually have a different name)

I think this would be the best option, rather than having different "types" of unclassified, since the only thing that is separating anything here is the fact that the Jje are carriable to have an "interactive" mechanic with the Ayym. If I had to think back before I created these and if we thought about this then, we'd probably just have the "Unclassified" category for all of these, and then I would have to give something different to the Jje anyway. Since this is essentially file-based, having "Unclassified " would work and be consistent, in my opinion, rather than having an irregular second category that applies to one other (current) example.

warp-core commented:
Making a category name that looks like a typo is probably not a good idea.

I'm having difficulty understanding the opposition to the presence of "(carried)" given that carriability seems to be the key differentiator between the ships that are being assigned to each category.

> minor doesn't specify anything clearly, although it allows us all to assume various things.

Why is desirable?
It sounds like you're saying you want the name to have whatever meaning the reader wants to assign to it? That doesn't seem like a particularly useful thing to do.

Zitchas commented:
I don't like "carried" because that very explicitly states that it is akin to a fighter. The whole point of this is to label them as things we don't understand, so anything that establishes a baseline comprehension is undesirable. We don't know if these are immature members of the species, or parasites that cling to it, or specialist bred minions to defend itself with. And more to the point, we definitely don't know what future carried unclassified ships might be. There are a lot of uses for carried things, with the classic fighter-like entity is only one.

warp-core commented:
> The whole point of this is to label them as things we don't understand

This is a very strange approach to assigning labels that never appear in game. Your description even states:

> This is generally not a directly in-game visible change.

While, in character, these things may indeed not have any meaningful categorisation compared to other ships, these categorisations don't appear to serve any in-character purpose.

The only good reasons for changing the categories of any of these ships are out of character ones. And, out of character, we know exactly that the differentiating factor between these two categories is that one of them is carryable and the other is not.

Saugia commented:
> And more to the point, we definitely don't know what future carried unclassified ships might be. There are a lot of uses for carried things, with the classic fighter-like entity is only one.

We don't add things to the game for the sake of the *possibility* that something might be in the future; I actually remember some discussion prior that you mentioned we should worry about what's already in the game. Currently we only have so many of these "alien ships" - if there were as many aliens as human ships in the game, then I would be convinced a purpose to have a a secondary class for them, as we do with L/M/H warships. As it currently stands, there only around 10 that fit this, with a few extra being game mechanics, which are more than well-covered by a single category.

Again, categories are purpose/use based, *not* lore based. I think it would be completely acceptable that within missions or discussions that we dive into the biology of a lifeform, but for something that is specifically used as an organizational tool that isn't seen in-game, it only needs to be used for that.

That said, since this often occurs in the past, I don't think going in circles about this will do anything but have this PR sit open for months. The PR introduces a new class to place these odd examples in for organization. I see the acceptance of a brand-new class for organization a plus and useful addition to the game's structure. Rather than argue about specific details and be stuck forever, I think we should be happy about the desire for this addition and move past it.

Zitchas commented:
> > This is generally not a directly in-game visible change.
> 
> While, in character, these things may indeed not have any meaningful categorisation compared to other ships, these categorisations don't appear to serve any in-character purpose.

As you are no doubt very well aware, people read the data files, categories get put on ships in wikis, and people take those categories as lore. They might not be intended as lore, but intentions are pretty much irrelevant when someone not party to the development process reads "fighter" as the category in the wiki and takes it as in-game fact. Given that this is the case, we might as well label it something that contributes to the lore instead of something that directly goes against it.

ziproot commented:
I proposed carried because that’s the distinguishing factor here. The category only exists because there needs to be a separate one for ships that are carried, with “carried” being a game specific term involving a ship being able to dock to a different ship. The carried immediately shows anyone looking at the file that this ship is unclassified and is carried, as opposed to the other unclassified ships that cannot dock.

This is in comparison to, say, unclassified minor, or unclassified, which doesn’t just refer to the carried ships. Again, I would be fine with unclassified by itself as well, if people don’t think that would be too confusing, but I think this makes the most sense to prospective viewers of the file who want to know why the category exists.

Of course, I made my suggestions a comment and not a change request because it doesn’t matter too much here. If leaving the PR the way it is is how this gets merged, fine.

MasterOfGrey commented:
> As you are no doubt very well aware, people read the data files, categories get put on ships in wikis, and people take those categories as lore. They might not be intended as lore, but intentions are pretty much irrelevant when someone not party to the development process reads "fighter" as the category in the wiki and takes it as in-game fact. Given that this is the case, we might as well label it something that contributes to the lore instead of something that directly goes against it.

Since the categories got ported to data files it's entirely possible to have other categories be carried that aren't fighters.

Arguably, given that, making it obvious that it's (carried) makes it *more* obtuse, since if it were genuinely similar to a fighter it would in fact be classified as "Fighter".

And it is not.

Using `Unclassified` and `Unclassified (Carried)` is being explicit about the fact that they're all unclassified and not understood. Text in brackets is very clearly not a matter of lore, but of game mechanics - since categories have to be defined to enable carried behaviour.
There is no conflict with the lore here - and arguably, the mystery is heightened by not implying some kind of status with `Minor`.

TheGiraffe3 commented:
What is the status on this PR?

TheGiraffe3 commented:
@Zitchas could you resolve conflicts and address comments?

Zitchas commented:
Conflicts have been resolved as of 2025-02-26.

Status is: Conflicted. As in, I hold one opinion, and other people hold other opinions, and being a data label, neither side has any factual grounds to say "this is right" or "this is wrong," so it comes down to opinions on what makes things better, and obviously at this point it should be clear that I think one way makes for a better game environment than the other people do. I have no interest in arguing over it, so there it sits.

As in, people are going to look at the data, and put things into the wiki. And have it become part of their internal lore. We can either fight against it, and try to tell everyone that happens to look at the wiki or data files that they're wrong, or we can use it  to create better lore. Likewise, "Unclassified " is just begging to have someone file a typo report on it. Why fight against the way people think and play, when we could instead just clearly state. "This is an unclassified thing" and "This is a unclassified (minor) thing, aka a smaller thing framed in a more scientific term." The former is just inviting confusion and problems, "Why do we have two wiki entries for "Unclassified"?", the latter uses it to create something that rolls with it and is more interesting.

ziproot commented:
I had done Unclassified (Carried) which eight other people including at least three devs seem to support, then you specifically suggested having the space instead.

![image](https://github.com/user-attachments/assets/3092fda9-dbca-48a8-a86b-cc133b81456e)

ziproot commented on data/vyrmeid/vyrmeid.txt:
```diff
@@ -77,7 +77,7 @@ ship "Vyrmeid" "Vyrmeid (B)"
 	"never disabled"
 	attributes
-		category "Fighter"
+		"category" "Unclassified Minor"
```
```suggestion
		"category" "Unclassified (Carried)"
```

ziproot reviewed (commented):
Update to use carried

Zitchas commented:
I just said there was a disagreement over what works best, I didn't say it was an even split or anything. I haven't even *thought* about this PR in months until someone specifically asked me to resolve the conflicts and address the comments. 

I have resolved the conflict, and addressed the comments. I am under no illusion that my addressing of them is satisfying to anyone, although hope springs eternal.

Zitchas commented on data/drak/indigenous.txt:
```diff
@@ -125,7 +125,7 @@ ship "Void Sprite" "Void Sprite (Infant)"
 	"never disabled"
 	attributes
-		category "Light Warship"
+		category "Unclassified Minor"
```
I don't get it. Because you really don't want the infant Void Sprites to ever be carriable?

ziproot replied:
The infant void sprites are not *currently* carry-able. If they ever become carry-able in the future, then this can be changed to Unclassified (Carried) like the others. Unclassified Minor confuses this with the other ships, when the entire purpose of having two (rather than one) categories is because of the carried ships.

Zitchas replied:
That actually supports my point. I want that lined blurred. If there's any other cases where this could be done, please point them out. 

The literal only point to having two separate categories is that the mechanics require it. So, since the mechanics require it, might as well make it contribute to lore describing things that are smaller, separable appendages, or other unknown things.

Zitchas replied:
As an aside, I find it hilarious how dead set you (and the two devs that appear to care even slightly) care about this. I think it matters, hence my pushing this for something that actually makes things better. Everyone else seems to think it doesn't matter, so why fight so hard on  it?

mOctave commented:
Hmm... how hard would it be to implement display name for categories? Then we could have `Unclassified` and `Unclassified (Carried)` for the proper names, but on the off chance they're seen in game they could all just appear under `Unclassified`.

warp-core commented:
> Hmm... how hard would it be to implement display name for categories? Then we could have `Unclassified` and `Unclassified (Carried)` for the proper names, but on the off chance they're seen in game they could all just appear under `Unclassified`.

Wouldn't that result in ships having bays for a category of ship named "Unclassified" but not being able to actually carry other ships that claim to belong to that category?

Saugia commented:
It's been a minute since I've looked at this PR, but going through my last comments, I was in general for adding the two categories. I personally don't see it difficult one way or another how we approach this, be it either "minor" or "carried"; the main contention point is around the necessary designation of a carried ship. So, the question would be, what are these categories for? Are they for the mechanic? Or are they for the ship designation?

I think the minor is fine, at the very least much better than having a space after classified, which I agree that it just looks like a typo. However, as for the use of minor, I think it's appropriate that, since the entire purpose of having a second category is due to a game mechanic, that it should be applied specifically to carried alien ships. The Void Sprite infant is not a carried ship, so I don't think it should get it, or other "smaller unclassified ships, when again the entire purpose of the separation of the class is due to mechanics, not research nature. If the void sprite infants someday become carriable like that, then I'd agree to giving them that.

So, that puts a bit of a compromise. We move away from "carried" and call carried beings "minor", which can simply be seen as the equivalent of "fighter" for a being, and we only apply it to carried beings mechanically, since that's the only purpose of the second category. Does that sound fair?

mOctave commented:
> > Hmm... how hard would it be to implement display name for categories? Then we could have `Unclassified` and `Unclassified (Carried)` for the proper names, but on the off chance they're seen in game they could all just appear under `Unclassified`.
> 
> Wouldn't that result in ships having bays for a category of ship named "Unclassified" but not being able to actually carry other ships that claim to belong to that category?

Oh, yeah, I guess that would be a problem. Although I think the debate with Unclassified ships was that you're never supposed to actually see them in-game anyways?

Zitchas commented:
> So, that puts a bit of a compromise. We move away from "carried" and call carried beings "minor", leaning more into the ship's proper classification, and which can simply be seen as the equivalent of "fighter" for a being/unclassified entity, and we only apply minor to carried beings mechanically, since that's the only purpose of the second category. Does that sound fair?

That is reasonable. 

I have removed the "Minor" from the Void Sprite Infant.

ziproot reviewed (approved):
Perfect is the enemy of good enough. I would still prefer it to say carried instead, but I never intended for this to be a blocker, so switching to an approval.

TheGiraffe3 reviewed (changes requested):
There are also animals added by the Incipias and Avgi PRs that still need to be updated here.

TheGiraffe3 commented:
To be specific, the changes should be made to ships:
```
ship "Windjammer"
ship "Bluejacket"
ship "Pollen"
ship "Sprout"
ship "Tace Mesa"
ship "Pincer Beast"
```

Zitchas commented:
Thanks for pointing those out. They have now been converted.

TheGiraffe3 reviewed (approved):
I think this conceptually makes a lot of sense, and it definitely does add a touch of mystery to these. Thank you for working on it.

Zitchas commented:
Thank you, and you are welcome.

Saugia commented:
While I think it does read a bit silly now to have "Unclassified Minor" as a bay type, it's only used for these very few strange alien cases (3). The overall "Unclassified" role though is acceptable and I do think it's a bit nicer than just calling something a Heavy Warship or such. Game Mechanics is also a bit more helpful on the data side.

Tested it and the ships work properly with the new bay definitions. 👍
